backend/src/main/java/com/google/step/utils/UserDatastoreUtil.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.utils;
+
+import com.google.appengine.api.datastore.DatastoreFailureException;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Key;
+
+
+/**
+ * This class represents a UserAccount in datastore and provides utility methods
+ * for working with Appengine User objects and Entities of kind "UserAccount".
+ * This class is designed to be the single point of responsibility for storing and reading
+ * UserAccount entities from the datstore.
+ */
+public final class UserDatastoreUtil {
```

Comment:
Is this only imported in UserAuthenticationUtil.java? If so might want to consider making this `protected`.

Reply:
I think I'm going to go ahead and keep them separate while I work on implementing the rest of the data objects (forms and campaigns) before deciding whether to keep them separate or merge them.
Also, I am anticipating needing to use this DatastoreUtil in other classes so for now I am going to keep it public.